WebViewControl.Avalonia/BaseControl.cs

```diff
@@ -6,24 +6,20 @@
 namespace WebViewControl {
 
     public abstract class BaseControl : Control {
-
+        private Window attachedWindow;
+        
         protected abstract void InternalDispose();
 
         protected override void OnAttachedToLogicalTree(LogicalTreeAttachmentEventArgs e) {
-            if (e.Root is Window window) {
-                // need to subscribe the event this way because close gets called after all elements get detached
-                window.AddHandler(Window.WindowClosedEvent, (EventHandler<RoutedEventArgs>)OnHostWindowClosed);
+            if (e.Root is Window window && attachedWindow != window) {
+                attachedWindow?.RemoveHandler(Window.WindowClosedEvent, (EventHandler<RoutedEventArgs>)OnHostWindowClosed);    
```

Comment:
This new logic means that we are just removing the previous handler when we are attaching a new window to the tree?

Reply:
Yes. I don't fully like it because does not allow to reuse the Control unless we first attach it to a new window.

It may be an issue in this use case:

1. Detach from logical tree
2. Close current window
3. Attach to new window **(This is not possible)**

We are forced to do it in this order:

1. Detach from logical tree
2. Attach to new window **(This is not possible)**
3. Close previous window

Reply:
I looked at the Avalonia source and when the detached is called it seems that we can look at some properties to infer that the window is about to close, namelly PlatformImpl and IsVisible

Reply:
<img width="948" alt="image" src="https://github.com/user-attachments/assets/1bb217ec-8d9d-46c7-934f-8484c72e6773">